docs/k8s/user-guide.mdx

```diff
@@ -464,3 +495,194 @@ spec:
                 port:
                   number: <service-port>
 ```
+
+# TCP and TLS Edges {#tcp-tls-edges}
+
+ngrok offers [TCP](https://ngrok.com/docs/tcp/) and
+[TLS](https://ngrok.com/docs/tls/) Edges which can be used to
+provide ingress to TCP or TLS based services. Both are implemented as CRDs and
+function similarly in broad strokes, albeit with slightly different
+configuration options offered. [Their CRD reference](/docs/k8s/crds#tcp-edges) is a
+useful companion to this guide.
+
+## (TLS Only) Get a Domain
+
+At least one `hostports` must be specified when creating a TLSEdge resource,
+which takes the form `<fqdn>:443`. The fully qualified domain name must first be
+reserved either via the ngrok dashboard or the [Domain](/docs/k8s/crds#domains) CRD.
+
+Example:
+
+```yaml
+apiVersion: ingress.k8s.ngrok.com/v1alpha1
+kind: Domain
+metadata:
+  name: tlsedgetest-ngrok-app
+spec:
+  domain: tlsedgetest.ngrok.app
+```
+
+## Create the Edge
+
+Create the edge CRD. These resources are fairly similar, and both require you to
+specify a [TunnelGroupBackend](/docs/k8s/crds#tunnelgroupbackend). This consists of a
+list of labels that determine which specific [Tunnel](/docs/k8s/crds#tunnels) should
+receive traffic from the edge. Both may also specify [IP
+Policies](https://ngrok.com/docs/tls/ip-restrictions/) for limiting access
+to the edge. At the time of writing, these policies must be provided as a
+reference in the form `ipp_<id>`.
+
+On top of the options available to TCP Edges, TLS Edges support (and require) a few other options:
+
+- (required) `hostports`: A list of `"<fqdn>:443"` strings declaring the list of
+  reserved domains for the edge to listen on.
+- [`tlsTermination`](https://ngrok.com/docs/api/resources/tls-edge-tls-termination-module/): Configure the TLS Termination behavior. The `terminateAt` field may be set to `upstream` to pass the encrypted stream to the Tunnel backend, or `edge` to terminate the TLS stream at the ngrok edge, and pass plaintext bytes to the Tunnel.
+- [`mutualTls`](https://ngrok.com/docs/api/resources/tls-edge-mutual-tls-module/): Configure client certificate validation at the edge. Requires a reference to a [Certificate Authority](https://ngrok.com/docs/api/resources/certificate-authorities/).
+
+TCP Example:
+
+```yaml
+apiVersion: ingress.k8s.ngrok.com/v1alpha1
+kind: TCPEdge
+metadata:
+  name: test-edge
+spec:
+  backend:
+    labels:
+      app: tcptestedge
+```
+
+Because TCP Edges don't currently support providing a reserved TCP address. On edge creation, one will be allocated for them, and will be visible by checking the status of the resource:
```

Comment:
```suggestion
TCP Edges currently do not support providing a reserved TCP address. Therefore, one will be allocated upon edge creation. This can be viewed by checking the status of the resource:
```